xcdat/spatial_avg.py

```diff
@@ -324,93 +324,152 @@ def _get_weights(
         """
         BoundsByType = TypedDict(
             "BoundsByType",
-            {"domain": Optional[xr.DataArray], "region": Optional[RegionAxisBounds]},
+            {"domain": xr.DataArray, "region": Optional[RegionAxisBounds]},
         )
+
         bounds: Dict[str, BoundsByType] = {
             "lat": {
-                "domain": self._dataset.bounds.get_bounds("lat"),
+                "domain": self._dataset.bounds.get_bounds("lat").copy(),
```

Comment:
By adding `.copy()` here, we don't need to add `.copy()` in proceeding methods that use this value.